serverlessworkflow/modules/ROOT/pages/integrations/camel-routes-integration.adoc

```diff
@@ -0,0 +1,133 @@
+
+{context} can integrate with link:{camel_url}[Apache Camel Routes] by adding the Kogito Quarkus Camel Add-on to your project. It enables the workflow engine to identify and call Camel routes declared in YAML or XML in the same workflow project context.
+
+[[proc-enable-quarkus-camel]]
+== Enabling Quarkus Camel in {context}
+
+You can enable Quarkus Camel in your project.
+
+.Prerequisites
+* A workflow application is created. 
++
+For more information about creating a workflow, see xref:getting-started/create-your-first-workflow-service.adoc[Creating your first workflow service].
+
+.Procedure
+. To add the Quarkus Camel to your workflow application, add the `org.kie.kogito:kogito-addons-quarkus-camel` dependency to the `pom.xml` file of your project as follows:
++
+--
+.Dependency to be added to the `pom.xml` file to enable metrics
+[source,xml]
+----
+<dependency>
+    <groupId>org.kie.kogito</groupId>
+    <artifactId>kogito-addons-quarkus-camel</artifactId>
+</dependency>
+----
+--
+
+[[con-creating-camel-routes]]
+== Creating Camel routes in {context}
+
+You can add YAML or XML Camel routes to your workflow project.
+
+.Procedure
+. Create a YAML or XML Camel Routes using your IDE or the link:{kaoto_url}[Kaoto VSCode Editor] and place them in the `src/main/resources/routes` directory.
+. The route `from` endpoint must be a `direct` component. That's the endpoint producer expected by the workflow engine.
+. The route response must be in a valid format that the workflow context can understand:
++
+include::../../pages/_common-content/camel-valid-responses.adoc[]
++
+The response will be merged into the workflow state context. If it is an array or a complex object, the response will be added to the special attribute `response`.
+
+.Example of a Camel route with the `direct` endpoint returning a valid JSON string representation
+[source,yaml]
+----
+- from:
+    uri: direct:logRouteReplaceHeader
+    steps:
+      - setBody:
+          simple: '{ "id": "${header.WorkflowID}", "arg1": { "arg2": "value1" } }'
+      - log:
+          message: We received the ${body}
+----
+
+[[con-importing-camel-routes]]
+== Defining and referencing Camel functions in the Workflow DSL
+
+You can define and reference your Camel functions in the workflow definition.
+
+.Prerequisites
+* You have created Camel routes in the workflow Maven project.
+
+.Procedure
+. In the `functions` definition section of your workflow DSL, declare the Camel route as exemplified below:
++
+.Example of a Camel Route function definition
+[source,json]
+----
+{
+  "functions": [
+    {
+      "name": "logRoute",
+      "type": "custom",
+      "operation": "camel:direct:logRouteReplaceHeader"
+    }
+  ]
+}
+----
++
+The operation description must have the prefix `camel:direct:`, indicating that you want to produce a message to this route via the link:{camel_extensions_url}/direct.html[Camel Direct Component]. Direct is the only component supported by {context} at the moment.
```

Comment:
We should have a page with the Camel supported components instead of writing it into the text, which we may forget about ...

Reply:
We won't support anything else. If so, we can then update the guide with a supported components.

Reply:
ok